third_party/2and3/flask/app.pyi

```diff
@@ -0,0 +1,139 @@
+from .ctx import AppContext, RequestContext, _AppCtxGlobals
+from .globals import _request_ctx_stack, g, request, session
+from .helpers import _PackageBoundObject, find_package, get_debug_flag, get_env, get_flashed_messages, get_load_dotenv, locked_cached_property, url_for
+from .logging import create_logger
+from .sessions import SecureCookieSessionInterface
+from .signals import appcontext_tearing_down, got_request_exception, request_finished, request_started, request_tearing_down
+from .templating import DispatchingJinjaLoader, Environment
+from .wrappers import Request, Response
+from .testing import FlaskClient
+from typing import Any, Callable, ContextManager, Dict, List, Optional, Type, TypeVar, Union
+
+def setupmethod(f: Any): ...
+
+_T = TypeVar('_T')
+
+class Flask(_PackageBoundObject):
+    request_class: type = ...
+    response_class: type = ...
+    jinja_environment: type = ...
+    app_ctx_globals_class: type = ...
+    config_class: type = ...
+    testing: Any = ...
+    secret_key: Any = ...
+    session_cookie_name: Any = ...
+    permanent_session_lifetime: Any = ...
+    send_file_max_age_default: Any = ...
+    use_x_sendfile: Any = ...
+    json_encoder: Any = ...
+    json_decoder: Any = ...
+    jinja_options: Any = ...
+    default_config: Any = ...
+    url_rule_class: type = ...
+    test_client_class: type = ...
+    test_cli_runner_class: type = ...
+    session_interface: Any = ...
+    import_name: str = ...
+    template_folder: str = ...
+    root_path: Any = ...
+    static_url_path: Any = ...
+    static_folder: str = ...
+    instance_path: Any = ...
+    config: Dict[str, Any] = ...
+    view_functions: Any = ...
+    error_handler_spec: Any = ...
+    url_build_error_handlers: Any = ...
+    before_request_funcs: Dict[Optional[str], List[Callable[[], Any]]] = ...
+    before_first_request_funcs: List[Callable[[], None]] = ...
+    after_request_funcs: Dict[Optional[str], List[Callable[[Response], Response]]] = ...
+    teardown_request_funcs: Dict[Optional[str], List[Callable[[Optional[Exception]], Any]]] = ...
+    teardown_appcontext_funcs: List[Callable[[Optional[Exception]], Any]] = ...
+    url_value_preprocessors: Any = ...
+    url_default_functions: Any = ...
+    template_context_processors: Any = ...
+    shell_context_processors: Any = ...
+    blueprints: Any = ...
+    extensions: Any = ...
+    url_map: Any = ...
+    subdomain_matching: Any = ...
+    cli: Any = ...
+    def __init__(self, import_name: str, static_url_path: Optional[str] = ..., static_folder: str = ..., static_host: Optional[str] = ..., host_matching: bool = ..., subdomain_matching: bool = ..., template_folder: str = ..., instance_path: Optional[str] = ..., instance_relative_config: bool = ..., root_path: Optional[str] = ...) -> None: ...
+    def name(self) -> str: ...
+    @property
+    def propagate_exceptions(self): ...
+    @property
+    def preserve_context_on_exception(self): ...
+    def logger(self): ...
```

Comment:
I this actually a property? There are dozens of errors like:
```
error: "Callable[[], Any]" has no attribute "info"
error: "Callable[[], Any]" has no attribute "error"
error: "Callable[[], Any]" has no attribute "warning"
error: "Callable[[], Any]" has no attribute "exception"
```

Reply:
Indeed. Fixed.